envs/monkey_zoo/blackbox/test_blackbox.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+from time import sleep
+
+from envs.monkey_zoo.blackbox.island_client.monkey_island_client import MonkeyIslandClient
+from envs.monkey_zoo.blackbox.analyzers.communication_analyzer import CommunicationAnalyzer
+from envs.monkey_zoo.blackbox.island_client.island_config_parser import IslandConfigParser
+from envs.monkey_zoo.blackbox.utils import gcp_machine_handlers
+from envs.monkey_zoo.blackbox.tests.basic_test import BasicTest
+from envs.monkey_zoo.blackbox.log_handlers.test_logs_handler import TestLogsHandler
+
+DEFAULT_TIMEOUT_SECONDS = 5*60
+MACHINE_BOOTUP_WAIT_SECONDS = 30
+GCP_TEST_MACHINE_LIST = ['sshkeys-11', 'sshkeys-12', 'elastic-4', 'elastic-5', 'haddop-2-v3', 'hadoop-3', 'mssql-16',
+                         'mimikatz-14', 'mimikatz-15', 'final-test-struts2-23', 'final-test-struts2-24',
+                         'tunneling-9', 'tunneling-10', 'tunneling-11', 'weblogic-18', 'weblogic-19', 'shellshock-8']
+
+
+@pytest.fixture(autouse=True, scope='session')
+def GCPHandler(request):
+    GCPHandler = gcp_machine_handlers.GCPHandler()
+    GCPHandler.start_machines(" ".join(GCP_TEST_MACHINE_LIST))
+    wait_machine_bootup()
+
+    def fin():
+        GCPHandler.stop_machines(" ".join(GCP_TEST_MACHINE_LIST))
+
+    request.addfinalizer(fin)
+
+
+@pytest.fixture(autouse=True, scope='session')
+def delete_logs():
+    print("Deleting monkey logs before new tests.")
+    TestLogsHandler.delete_log_folder_contents()
+
+
+def wait_machine_bootup():
+    sleep(MACHINE_BOOTUP_WAIT_SECONDS)
+
+
+@pytest.fixture(scope='class')
+def island_client(island):
+    island_client_object = MonkeyIslandClient(island)
+    island_client_object.reset_env()
+    yield island_client_object
+
+
+@pytest.mark.usefixtures('island_client')
+# noinspection PyUnresolvedReferences
+class TestMonkeyBlackbox(object):
```

Comment:
I think we should randomize the order of tests to make sure we aren't flaky. See https://pypi.org/project/pytest-random-order/ for more details.

Reply:
ATM I'm getting internal errors when using pytest-random-order. Lets postpone this part till python 3 (updated packages should have no compatibility issues)

Reply:
ACK - remove from requirements